src/screens/Connectors/ConnectorMetaData/BankDebit/BankDebit.res

```diff
@@ -0,0 +1,181 @@
+    ~fieldsArray: array<ReactFinalForm.fieldRenderProps>,
+    ~paymentMethod: string,
+    ~paymentMethodType: string,
+    ~getConnectorId: string => string,
+  ) => {
+    open LogicUtils
+    let (currentSelection, setCurrentSelection) = React.useState(_ => "")
+
+    let enabledList = (
+      fieldsArray->Array.get(0)->Option.getOr(ReactFinalForm.fakeFieldRenderProps)
+    ).input
+
+    let input: ReactFinalForm.fieldRenderPropsInput = {
+      name: "string",
+      onBlur: _ => (),
+      onChange: ev => {
+        let value = ev->Identity.formReactEventToString
+        let getPaymentMethodsObject = connector => {
+          open BankDebitTypes
+          {
+            payment_method: paymentMethod,
+            payment_method_type: paymentMethodType,
+            connector_name: connector,
+            mca_id: getConnectorId(connector),
+          }
+        }
+        if value->LogicUtils.isNonEmptyString {
+          let paymentMethodsObject = value->getPaymentMethodsObject
+          setCurrentSelection(_ => value)
+
+          let existingPaymentMethodsArray =
+            enabledList.value->getArrayDataFromJson(BankDebitUtils.itemToObjMapper)
+
+          let newPaymentMethodsArray =
+            existingPaymentMethodsArray->Array.filter(item =>
+              item.payment_method_type !== paymentMethodType
+            )
+
+          newPaymentMethodsArray->Array.push(paymentMethodsObject)
+          enabledList.onChange(newPaymentMethodsArray->Identity.anyTypeToReactEvent)
+        }
+      },
+      onFocus: _ => (),
+      value: currentSelection->JSON.Encode.string,
+      checked: true,
+    }
+    <SelectBox.BaseDropdown
+      allowMultiSelect=false
+      buttonText="Select Value"
+      input
+      options
+      hideMultiSelectButtons=false
+      showSelectionAsChips=true
+      customButtonStyle="w-full"
+      fullLength=true
+      dropdownClassName={`${options->PaymentMethodConfigUtils.dropdownClassName}`}
+    />
+  }
+}
+
+@react.component
+let make = (
+  ~onCloseClickCustomFun,
+  ~setShowWalletConfigurationModal,
+  ~update,
+  ~paymentMethod,
+  ~paymentMethodType,
+  ~initialValues,
+  ~setInitialValues,
+) => {
+  open LogicUtils
+  open BankDebitUtils
+  let connectorList = HyperswitchAtom.connectorListAtom->Recoil.useRecoilValueFromAtom
+  let formState: ReactFinalForm.formState = ReactFinalForm.useFormState(
+    ReactFinalForm.useFormSubscription(["values"])->Nullable.make,
+  )
+  let form = ReactFinalForm.useForm()
+
+  let connectorsListPMAuth =
+    connectorList->ConnectorUtils.getProcessorsListFromJson(
+      ~removeFromList=ConnectorTypes.PMAuthenticationProcessor,
+    )
+  let pmAuthConnectors =
+    connectorsListPMAuth->Array.map(item => item.connector_name)->removeDuplicate
+
+  let options = pmAuthConnectors->dropdownOptions
+
+  let getConnectorId = connector => {
+    let connectorData = connectorsListPMAuth->Array.find(item => {
+      item.connector_name == connector
+    })
```

Comment:
Is there any particular reason we are using `connector name as the string` type and not `connector as enum` defined

Reply:
we can also use connector as enum defined,